firewood/src/storage/linear/mod.rs

```diff
@@ -71,44 +68,15 @@ mod proposed;
 #[cfg(test)]
 mod tests;
 
-#[derive(Debug)]
-pub(super) struct LinearStore<S: ReadLinearStore> {
-    state: S,
-}
-
 /// All linearstores support reads
-pub(super) trait ReadLinearStore: Send + Sync + Debug {
```

Comment:
Unfortunately, you should not remove this. If you run against the nightly version of clippy, it will complain that you could have a `ReadLinearStore` that is not `Send` or `Sync`, and since you wrap that in an `Arc` (and we probably will need these to be Send/Sync) we should specifically say that this is required.